elyra/pipeline/processor_kfp.py

```diff
@@ -209,7 +209,10 @@ def process(self, pipeline):
             # Upload the compiled pipeline, create an experiment and run
 
             try:
-                description = f"Created with Elyra {__version__} pipeline editor using '{pipeline.source}'."
+                if pipeline.description is not None:
+                    description = pipeline.description
+                else:
+                    description = f"Created with Elyra {__version__} pipeline editor using `{pipeline.source}`."
```

Comment:
I'm curious why you didn't take the same approach as the other two locations and pre-assign the variable to avoid the `else` block?

Hmm - since the same string is generated when `pipeline.description` is not set and since both values (`__version__` and `pipeline.source`) are available from the property accssor, should the property accessor always return a value and we can eliminate these three locations altogether?

Reply:
> I'm curious why you didn't take the same approach as the other two locations and pre-assign the variable to avoid the else block?

No reason, but I've changed it now to be consistent. I also renamed the variable to be consistent.

> Hmm - since the same string is generated when pipeline.description is not set and since both values (__version__ and pipeline.source) are available from the property accessor, should the property accessor always return a value and we can eliminate these three locations altogether?

I hadn't thought about that. Our original intention was to always provide a "description" that would enable the user to associate a DAG in the runtime environment with a pipeline. Since descriptions were not supported at that time we decided to use the static string instead. If we were to alter the getter to return the static string this might create issues in the future should other pieces of code start utilizing `Pipeline` objects. 

For example, one could argue that users should always provide a pipeline description. If pipeline validation were to use the `Pipeline` class, it would not be able to determine whether a description was provided by the user and thus not produce a [best practice] hint.

Reply:
This particular `Pipeline` class is downstream from validation and represents the _runtime representation_ of the pipeline and contains `Operation` instances, rather than _nodes_.  The `Pipeline` class you're referring to (:smile: ) is the `Pipeline` class contained within the `PipelineDefinition` hierarchy that @lresende is introducing with #2082.  I don't think there are plans to utilize `PipelineDefinition.Pipeline` where `Pipeline.Pipeline` instances are used today (and certainly not the reverse).  As a result, we're past the point of needing to detect whether or not a (future) required value is missing.